cdc/processor/pipeline/actor_node.go

```diff
@@ -0,0 +1,85 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package pipeline
+
+import (
+	"context"
+
+	"github.com/pingcap/errors"
+	"github.com/pingcap/tiflow/pkg/pipeline"
+)
+
+// ActorNode is an async message process node, it fetches and handle table message non-blocking
+// if processing is blocked, the message will be cached and wait next run
+type ActorNode struct {
+	messageStash     *pipeline.Message
+	parentNode       AsyncMessageHolder
+	messageProcessor AsyncMessageProcessor
+}
+
+// NewActorNode create a new ActorNode
+func NewActorNode(parentNode AsyncMessageHolder, messageProcessor AsyncMessageProcessor) *ActorNode {
+	return &ActorNode{
+		parentNode:       parentNode,
+		messageProcessor: messageProcessor,
+	}
+}
+
+// TryRun get message from parentNode and handle it util there is no more message to come
+//  or message handling is blocking
+// only one message will be cached
+func (n *ActorNode) TryRun(ctx context.Context) error {
+	for {
+		// batch?
+		if n.messageStash == nil {
+			n.messageStash = n.parentNode.TryGetDataMessage()
+		}
```

Comment:
Will it be a busy loop?

Reply:
if can not get message or can not process message ,  will break the for loop